### ISSUE TITLE
vscode extension: render button disappeared (replaced by preview)

### ISSUE THREAD

**Issue by driftwoods** [quarto-dev/quarto#276]
The render button is replaced by preview button in the latest update of quarto extension (version 1.100.0 released on Sep 24, 2023) in vscode. Users can't select output format (pdf or html) with preview button. This happened to vscode in both windows and Linux.

**Comment by jjallaire** [comment]
There is now a **Quarto: Preview Format** command that handles selecting a format to preview: https://quarto.org/docs/tools/vscode.html#render-and-preview

A major benefit of this command over the previous one is that it dynamically discovers the formats actually declared in the YAML and presents a list of those (the previous one was hard coded to PDF, Docx, etc.)

**Comment by driftwoods** [comment]
The problem is that Quarto:Preview only output HTML format. I can't specify pdf output using Quarto: Preview Format, like the one shown in the link you provided.

**Comment by jjallaire** [comment]
Right, you would need to add `pdf` as an explicit format within the document options for it to show up in the list. e.g.

```yaml
---
format:
   html: default
   pdf: default
---
```

(or instead of `default` specify various format-specific options)

**Comment by jjallaire** [comment]
It should indeed be consulting the project file (and does when I test it locally). Could you provide a link to a repo with a project that exhibits this behavior?

The problem with the old behavior is that it is completely static (VS Code makes us pre-allocate the commands / menu names) which means it doesn't work at all with custom formats (or any format that isn't PDF or docx). Note that once you've done **Quarto: Preview Format** the **Quarto: Preview** command is now bound automatically to that format. So in a session where you preview e.g. 15 times you now have 16 steps rather than 15, not a terrible overhead :-)
